tdp_core/settings/model.py

```diff
@@ -111,14 +111,14 @@ class GlobalSettings(BaseSettings):
     def is_development_mode(self) -> bool:
         return self.env.startswith("dev")
 
-    def get_nested(self, key: str, default: Any = None) -> Union[Any, None]:
+    def get_nested(self, key: str, default: Optional[Dict] = None) -> Dict:
         # TODO: Set deprecated
         keys = key.split(".")
         plugin_id = keys[0]
         dic = self.dict(include={plugin_id})
         for key in keys:
             dic = dic.get(key, None) if dic else None
-        return dic if dic is not None else default
+        return dic or default or {}
```

Comment:
`dic if dic is not None else default` is a very different to `dic or ...`, as I explicitly want to allow falsy values like 0 or `False`. 
I think you somehow expect this function to always return a dict, or am I missing something? This function can return anything really, as long as it is a JSON compatible datatype.

Reply:
Every usage I've encountered is getting a "sub config" (dict), which is also what I'd assume a "settings.get_nested()" would do. Get some settingS, plural, of a nested kind.

And I've encountered usage (in this codebase) that looks sane when observed
```
_conf = manager.settings.get_nested("tdp_core.security.store.
...
if _conf.get("enable", False):
```
But explodes in your face, because the _conf is None.

And in general, the only code where it makes a difference is where dic would be a false-ish value (False, 0), it would return a {} in the end, which is treated as False by any `if/not` anyways.

And with the current types, you'd get a warning for using it wrongly, whereas `Any | None` is bascially the same as having no types, and using proper coding style, you'd always have to check if the _conf is None or if the value in the _conf is None.

In general, returning Optional values and Any is considered bad practice, as the user has to guess what he'll get back and should (but hardly ever will) check his result.

Reply:
@dvmarkusvogl unfortunately, not all use cases are like this. Feel free to take a look at all repos if you need convincing. 

The point is that it returns a specific key (subsettings or **leaf**) in the JSON representation of the settings, which is 1:1 how the config worked before. I added this function only as compatibility layer, and compatibility layers should ideally have no breaking change, because that's kind of the purpose 🤗

So now, we can't change the behaviour so significantly, but it seems your issue can easily be solved by just passing a sensible default, i.e. {} if you expect a Dict. The entire point of the function is that I (as a tdp_core maintainer) don't know what you will get. 

So either pass the dict as your default, or even better, use the non-legacy pydantic model approach directly like settings.tdp_core.mongo.host for a typed, modern, recommended way. What is stopping you from using/updating to the latter approach?